src/v/kafka/server/tests/alter_config_test.cc

```diff
@@ -1368,6 +1373,82 @@ FIXTURE_TEST(test_unlicensed_alter_configs, alter_config_test_fixture) {
           failure);
     }
 
+    // Specific tests for schema id validation subject name strategy
+    {
+        using sns = pandaproxy::schema_registry::subject_name_strategy;
+
+        const auto full_validation = props_t{
+          with(kafka::topic_property_record_key_schema_id_validation, true),
+          with(kafka::topic_property_record_value_schema_id_validation, true),
+        };
+        test_cases.emplace_back(
+          "set_key_sns",
+          full_validation,
+          alter_props_t{
+            {set(
+              kafka::topic_property_record_key_subject_name_strategy,
+              sns::topic_name)},
+          },
+          failure);
+        test_cases.emplace_back(
+          "set_value_sns",
+          full_validation,
+          alter_props_t{
+            {set(
+              kafka::topic_property_record_value_subject_name_strategy,
+              sns::topic_name)},
+          },
+          failure);
+
+        const auto validation_with_strat = props_t{
+          with(kafka::topic_property_record_key_schema_id_validation, true),
+          with(kafka::topic_property_record_value_schema_id_validation, true),
+          with(
+            kafka::topic_property_record_key_subject_name_strategy,
+            sns::topic_name),
+          with(
+            kafka::topic_property_record_value_subject_name_strategy,
+            sns::topic_name),
+        };
+        test_cases.emplace_back(
+          "change_key_sns",
+          validation_with_strat,
+          alter_props_t{
+            {set(
+              kafka::topic_property_record_key_subject_name_strategy,
+              sns::record_name)},
+          },
+          failure);
+        test_cases.emplace_back(
+          "change_value_sns",
+          validation_with_strat,
+          alter_props_t{
+            {set(
+              kafka::topic_property_record_value_subject_name_strategy,
+              sns::record_name)},
+          },
+          failure);
+        test_cases.emplace_back(
+          "remove_key_sns",
+          validation_with_strat,
+          alter_props_t{
+            {remove(kafka::topic_property_record_key_subject_name_strategy)}},
+          success);
+
+        test_cases.emplace_back(
+          "remove_value_sns",
+          validation_with_strat,
+          alter_props_t{
+            {remove(kafka::topic_property_record_value_subject_name_strategy)}},
+          success);
+    }
+
+    // NOTE(oren): w/o schema validation enabled at the cluster level, related
+    // properties will be ignored on the topic create path. stick to COMPAT here
+    // because it's a superset of REDPANDA.
+    lconf().enable_schema_id_validation.set_value(
+      pandaproxy::schema_registry::schema_id_validation_mode::compat);
```

Comment:
We don't need to set this on all shards because this fixture test is supposed to run on a single shard, right? 

nit: I'm wondering if we should assert that here.

Reply:
It might be worth adding a helper to the `redpanda_thread_fixture` for setting config.

Elsewhere in the file it's done "more correctly":
```c++
    cluster::config_update_request r{
      .upsert = {{"enable_schema_id_validation", "compat"}}};
    app.controller->get_config_frontend()
      .local()
      .patch(r, model::timeout_clock::now() + 1s)
      .discard_result()
      .get();
```

Reply:
done

Reply:
I think you forgot to remove the old implementation here

Reply:
thx good catch, fixed

Reply:
oh lord 🤦 
yeah let's just do that